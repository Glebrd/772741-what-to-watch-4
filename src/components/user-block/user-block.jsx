import React from "react";
import {connect} from "react-redux";
import {getAuthorizationStatus, getUser} from "../../reducer/user/selectors";
import PropTypes from "prop-types";
import {userType} from "../../types";
import {Link} from "react-router-dom";
import {AuthorizationStatus} from "../../reducer/user/user";


const UserBlock = (props) => {
  const {user, authorizationStatus} = props;
  const {avatarURL} = user;
  let fullAvatarURL = `https://4.react.pages.academy${avatarURL}`;

  const userAvatarBlock =
    <Link to={`/mylist`}>
      <div className="user-block__avatar">
        <img src={fullAvatarURL} alt="User avatar" width="63" height="63"/>
      </div>
    </Link>;

  const userSignInLink =
    <Link to={`/login`} href="" className="user-block__link">Sign in</Link>;

  return (
    <div className="user-block">
      {authorizationStatus === AuthorizationStatus.AUTH
        ?
        userAvatarBlock
        :
        userSignInLink
      }
    </div>
  );
};

UserBlock.propTypes = {
  user: userType,
  authorizationStatus: PropTypes.string,
};

const mapStateToProps = (state) => ({
  user: getUser(state),
  authorizationStatus: getAuthorizationStatus(state),
});

export {UserBlock};

export default connect(mapStateToProps)(UserBlock);
