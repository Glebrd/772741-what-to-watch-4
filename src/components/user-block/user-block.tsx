import * as React from "react";
import {connect} from "react-redux";
import {getAuthorizationStatus, getUser} from "../../reducer/user/selectors";
import {UserType} from "../../types";
import {Link} from "react-router-dom";
import {AuthorizationStatus} from "../../reducer/user/user";
import {AppRoute} from "../../const";

const SERVER_URL = `https://4.react.pages.academy`;

interface Props {
    user: UserType;
    authorizationStatus: string;
}

const UserBlock: React.FunctionComponent<Props> = (props: Props) => {
  const {user, authorizationStatus} = props;
  const {avatarURL} = user;
  const fullAvatarURL = SERVER_URL + avatarURL;

  const userAvatarBlock =
    <Link to={AppRoute.MY_LIST}>
      <div className="user-block__avatar">
        <img src={fullAvatarURL} alt="User avatar" width="63" height="63"/>
      </div>
    </Link>;

  const userSignInLink =
    <Link to={AppRoute.LOGIN} href="" className="user-block__link">Sign in</Link>;

  return (
    <div className="user-block">
      {authorizationStatus === AuthorizationStatus.AUTH
        ?
        userAvatarBlock
        :
        userSignInLink
      }
    </div>
  );
};

const mapStateToProps = (state) => ({
  user: getUser(state),
  authorizationStatus: getAuthorizationStatus(state),
});

export {UserBlock};

export default connect(mapStateToProps)(UserBlock);
