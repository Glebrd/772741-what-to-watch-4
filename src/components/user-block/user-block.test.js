import React from "react";
import renderer from "react-test-renderer";
import {UserBlock} from "./user-block";

const onSignInClick = () => {};
const user = {
  avatarURL: `/wtw/static/avatar/2.jpg`,
};

it(`UserBlock component renders`, () => {
  const tree = renderer
    .create(
        <UserBlock
          user={user}
          onSignInClick={onSignInClick}
        />
    ).toJSON();

  expect(tree).toMatchSnapshot();
});
