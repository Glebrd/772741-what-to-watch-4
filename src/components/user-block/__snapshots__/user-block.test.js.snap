// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserBlock component renders 1`] = `
<div
  className="user-block"
>
  <div
    className="user-block__avatar"
  >
    <img
      alt="User avatar"
      height="63"
      src="https://4.react.pages.academy/wtw/static/avatar/2.jpg"
      width="63"
    />
  </div>
</div>
`;
